vcf_filtering
================

## How to query for SNPs

``` bash
bcftools query -f '%POS\n' <filename.vcf.gz> | wc -l
```

# General filtering

Starting number of SNPs **32,279,975**

## 1.Remove sites located within 5bp of any InDels.

``` bash
bcftools filter -g 5 --threads 20 -O z -o gatk.indel5bp.vcf.gz gatk.vcf.gz
```

Number of SNPs remaining **28,179,617**

## 2. Hard filtering

``` bash
bcftools filter -e 'QD<2 | QUAL<30 | SOR>3 | FS>60 | MQ<40 | MQRankSum<-12.5 | ReadPosRankSum<-8' gatk.indel5bp.vcf.gz > gatk.indel5b.hardfilter.vcf.gz  
```

Number of SNPs remaining **23,863,147**

## 3. Relatedness

Check for the presence of clones, siblings, or other close familial
relationships among the sequenced individuals based on pairwise kinship
coefficient estimated by vcftools `--relatedness2`

``` bash
vcftools --gzvcf gatk.indel5bp.hardfilter.vcf.gz --relatedness2 --out gatk.indel5bp.hardfilter  
```

There were no pairs of samples with relatedness values indicating close
kinship.

Number of SNPs remaining **23,863,147**

## 4. Site depth, missing genotypes, minor allele frequency, and genotype quailty

``` bash
vcftools --gzvcf gatk.indel5bp.hardfilter.vcf.gz \
--max-missing 0.9 --minQ 30 --min-meanDP 10 --max-meanDP 33 \
--minDP 3 --maf 0.05 --minGQ 20 --remove-filtered-geno-all --recode \
--recode-INFO-all --stdout | \
gzip > gatk.indel5bp.hardfilter.DP90gdp3gq30.vcf.gz
```

Number of SNPs remaining **10,466,114**

## 5. Monomorphic sites

``` bash
bcftools filter -e 'AC=0 || AC==AN' --threads 20 gatk.indel5bp.hardfilter.DP90gdp3gq30.vcf.gz |gzip > gatk.filtered.vcf.gz
```

SNPs remaining **10,466,114**

## 6. Inbreeding coefficient

SNPs with negative inbreeding coefficients \< -0.05 were removed.
Negative values of inbreeding coefficient can be a proxy for poor
mapping quailty

``` bash
bcftools filter -e 'InbreedingCoeff < -0.05' gatk.filtered.vcf.gz |gzip > gatk.filteredv2.vcf.gz 
```

SNPs remaining **9,958137**

## Excluding individual with high proportion of missing genotype

``` bash
vcftools --remove-indv CTON069_S13 --gzvcf gatk.filteredv2.vcf.gz --recode --recode-INFO-all --stdout | gzip > gatk.filteredv3.vcf.gz
```

Individuals remaing 82 of 83

``` bash
vcftools --gzvcf gatk.filteredv3.vcf.gz  --depth 
vcftools --gzvcf gatk.filteredv3.vcf.gz  --missing-indv 
vcftools --gzvcf gatk.filteredv3.vcf.gz  --SNPdensity 10000 
```

<img src="01.filtering_snps_files/figure-gfm/unnamed-chunk-1-1.png" width="1152" style="display: block; margin: auto;" />

**Figure 1:** The statistics of average coverage depth and percentage of
missing genotypes in all samples

<img src="01.filtering_snps_files/figure-gfm/unnamed-chunk-2-1.png" width="1152" style="display: block; margin: auto;" />

**Figure 2:** SNP distribution for all samples

#### Hard filtering criteria

#### Note1

No threshold or cut/off was done for **Inbreeding coefficient**. When
applied a large mount of SNPs were excluded from the analysis.
Additionally, deeper thought needs to be given on whether or not this
criteria is appropriate for *Acanthchromis polyacanthus*. *A.
polyacanthus* is a species that does not posses a pelagic larvae stage
and contributes to the raising of offspring via parent care. Offspring
are unlikely to leave the reef they were hatched on. Therefore, founder
effects during colonization events may have had a significant impact on
population genetic structure and as sea levels rose after the last
glacial period and populations became cut off from each other, gene flow
would have become highly restricted (if present at all). Therefore, it
may not be surprising that populations have higher inbreeding
coefficients than other ‘natural’ populations. Within this quailty
filtering steps no inbreeding coefficient threshold was applied due to
the unique life history traits of the species being studied.

#### Note2

THe figure below shows the hard-cut off values that were used during the
quailty hard-filtering step

``` bash
bcftools query -f '%QD\n' gatk.indel5bp.vcf.gz > QD_values.txt
bcftools query -f '%QUAL\n' gatk.indel5bp.vcf.gz > QUAL_values.txt
bcftools query -f '%SOR\n' gatk.indel5bp.vcf.gz > SOR_values.txt
bcftools query -f '%FS\n' gatk.indel5bp.vcf.gz > FS_values.txt
bcftools query -f '%MQ\n' gatk.indel5bp.vcf.gz > MQ_values.txt
bcftools query -f '%MQRankSum\n' gatk.indel5bp.vcf.gz > MQRankSum_values.txt
bcftools query -f '%ReadPosRankSum\n' gatk.indel5bp.vcf.gz > RPRS_values.txt
```

``` bash
bcftools query -f '%QD\n' gatk.filteredv3.vcf.gz > QD_values_filter.txt
bcftools query -f '%QUAL\n' gatk.filteredv3.vcf.gz > QUAL_values_filter.txt
bcftools query -f '%SOR\n' gatk.filteredv3.vcf.gz > SOR_values_filter.txt
bcftools query -f '%FS\n' gatk.filteredv3.vcf.gz > FS_values_filter.txt
bcftools query -f '%MQ\n' gatk.filteredv3.vcf.gz > MQ_values_filter.txt
bcftools query -f '%MQRankSum\n' gatk.filteredv3.vcf.gz > MQRankSum_values_filter.txt
bcftools query -f '%ReadPosRankSum\n' gatk.filteredv3.vcf.gz > RPRS_values_filter.txt
```

<img src="01.filtering_snps_files/figure-gfm/unnamed-chunk-3-1.png" width="1152" />
