0.2Demographic patterns
================

``` r
knitr::opts_knit$set(root.dir = "C:/Users/jc527762/OneDrive - James Cook University/PhD dissertation/Data/Chapter2_genetics")
```

# Observed and expected heterozygosity

We used plink2 to calculate heterozygosity per sample using the
following command

``` bash
plink2 --vcf gatk.filteredv3.vcf.gz --allow-extra-chr --het --bad-freqs
```

As part of this calculation PLINK provides the observed and expected
heterozygosity (Ho, He) as well as the inbreeding coefficient **F**
which is calculated as $1-\frac{H0}{He}$

<img src="02.demographic_patterns/figure-gfm/unnamed-chunk-4-1.png" width="672" /><img src="02.demographic_patterns/figure-gfm/unnamed-chunk-4-2.png" width="672" /><img src="02.demographic_patterns/figure-gfm/unnamed-chunk-4-3.png" width="672" />

In addition to filters outlined previously in section

[0.1filtering_snps](https://github.com/schmidte10/Chapter2_Genetics/blob/main/01.filtering_snps.md)

further filtering steps for were performed for population structure
analysis (PCA, ADMIXTURE) as follows:

1.To avoid potential errors from sequencing, we only included sites with
minor allele count greater than or equal to two.

Create a map file for *A. polyacanthus*.

``` bash
bcftools view -H gatk.vcf.gz | cut -f 1 | uniq | awk '{print $0"\t"$0 }' > str_analysis/SCH11429.DPg90gdp3gq30.chrom-map.txt
```
