---
title: "Population structure"
output:
  github_document:
    pandoc_args: --webtex
--- 

In addition to filters outlined previously in section [0.1filtering_snps](https://github.com/schmidte10/Chapter2_Genetics/blob/main/01.filtering_snps.md) further filtering steps for were performed for population structure analysis (PCA, ADMIXTURE) as follows: 

1.To avoid potential errors from sequencing, we only included sites with minor allele count greater than or equal to two.

2.Sites in violation of Hardy-Weinberg equilibrium were removed at p-value < 1e-4. 

Samples were not pruned for linkage disequilibrium due to a limited samples size 



Create a map file for _A. polyacanthus_.  

```bash 

vcftools --gzvcf gatk/gatk.indel5bp.hardfilter.vcf.gz\
--max-missing 0.5 --min-meanDP 10 --max-meanDP 30\
--minDP 3 --minGQ 40 --mac 2 --remove-filtered-geno-all --recode\
--recode-INFO-all --stdout |\
gzip > 05.2.Population_structure/gatk.filtered.relaxed_studywide.vcf.gz

vcftools --remove-indv CTON069_S13 --gzvcf 05.2.Population_structure/gatk.filtered.relaxed_studywide.vcf.gz\
--recode --recode-INFO-all --stdout | gzip > 05.2.Population_structure/gatk.filtered.relaxed_studywide_invariants.vcf.gz 

bcftools view -H gatk.filtered.relaxed_studywide_invariants.vcf.gz | cut -f 1 | uniq | awk '{print $0"\t"$0 }' > relaxed_studywide_invariants.chrom-map.txt
```
Next, we converted our vcf file to plink format which included sites with a minor allele count greater than or equal to two and excluded sites with a p-value for exact test for Hardy-Weinberg Equilibrium below 1e-4. 

Note that multialleleic variants are removed via the '--max-allele 2' argument

```bash 

plink2 --vcf ../data/gatk.filtered.relaxed_studywide.vcf.gz --out apolyacanthus --allow-extra-chr --max-alleles 2 --make-bed --set-missing-var-ids @:# --double-id --geno 0.1 --mac 2 --hwe 0.0001

plink2 --bfile apolyacanthus --indep-pairwise 50 10 0.1 --out apolyacanthus \
    --allow-extra-chr --bad-ld --set-missing-var-ids @:#

plink2 --bfile apolyacanthus --extract apolyacanthus.prune.in --recode ped --allow-extra-chr --out apolyacanthus_ldpruned  

```
```bash 
for K in {1..10} 
do
  admixture --cv=10 apolyacanthus_ldpruned.ped $K | tee log.${K}.out
  awk -v K=$K '$1=="CV"{print K, $4}' log.${K}.out >> CV.txt
done
```

